packages/camera/camera_android_camerax/test/capture_request_options_test.dart

```diff
@@ -0,0 +1,143 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:camera_android_camerax/src/camerax_library.g.dart';
+import 'package:camera_android_camerax/src/capture_request_options.dart';
+import 'package:camera_android_camerax/src/instance_manager.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'capture_request_options_test.mocks.dart';
+import 'test_camerax_library.g.dart';
+
+@GenerateMocks(
+    <Type>[TestCaptureRequestOptionsHostApi, TestInstanceManagerHostApi])
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  // Mocks the call to clear the native InstanceManager.
+  TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());
+
+  group('CaptureRequestOptions', () {
+    tearDown(() {
+      TestCaptureRequestOptionsHostApi.setup(null);
+      TestInstanceManagerHostApi.setup(null);
+    });
+
+    test('detached create does not make call on the Java side', () {
+      final MockTestCaptureRequestOptionsHostApi mockApi =
+          MockTestCaptureRequestOptionsHostApi();
+      TestCaptureRequestOptionsHostApi.setup(mockApi);
+
+      final InstanceManager instanceManager = InstanceManager(
+        onWeakReferenceRemoved: (_) {},
+      );
+
+      final List<(CaptureRequestKeySupportedType, Object?)> options =
+          <(CaptureRequestKeySupportedType, Object?)>[
+        (CaptureRequestKeySupportedType.controlAeLock, true),
+      ];
+
+      CaptureRequestOptions.detached(
+        requestedOptions: options,
+        instanceManager: instanceManager,
+      );
+
+      verifyNever(mockApi.create(
+        argThat(isA<int>()),
+        argThat(isA<Map<int, Object?>>()),
+      ));
+    });
+
+    test(
+        'create makes call on the Java side as expected for suppported non-null capture request options',
```

Comment:
This is testing the null case, not the non-null case right?

Reply:
Yes haha thanks!